### PR TITLE
Add properties for adding resources/settings files

### PR DESCRIPTION
**Pull request by drewnoakes**
Contributes to https://github.com/dotnet/project-system/issues/7373

The legacy project properties pages embedded the assembly resources and settings files editors in the "application designer". The new UI takes a property-centric view of the project, and delegates these editors to their own standalone experience.

To help users in this transition, we want a link action in the UI that does two things:

1. Create the file if it does not currently exist
1. Opens the file's editor

CPS PR https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/344654 adds link action handlers for resources and settings files. This change adds pages and properties to surface this to the user.

image

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7489)

### CODE REVIEW

**Comment by drewnoakes** [comment]
This should only be merged once the following have inserted:

- https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/344654
- https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/344681

**Review by jjmew — APPROVED** [review]
:shipit:

**Review by melytc — APPROVED** [review]

**Review by tmeschter — APPROVED** [review]

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ResourcesPropertyPage.xaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      xmlns="http://schemas.microsoft.com/build/2009/properties">
+
+  <Rule.DataSource>
+    <DataSource Persistence="ProjectFileWithInterception"
+                SourceOfDefaultValue="AfterContext"
+                HasConfigurationCondition="False" />
+  </Rule.DataSource>
+
+  <StringProperty Name="ResourcesPagePlaceholderDescription"
+                  DisplayName="Ignored"
```
Comment (MiYanni) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ResourcesPropertyPage.xaml:17]: Is this a special value to make it not use DisplayName? Interesting that something like empty string or excluding the attribute entirely wouldn't do the same thing.

**Review by MiYanni — APPROVED** [review]
So after this PR and the CPS PR, will there be a follow-up to add the link handlers? Right now, these are no-op handlers.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ResourcesPropertyPage.xaml (hunk 1 above)
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ResourcesPropertyPage.xaml:17]: No it's not special. Just capturing the fact that it's unused in any way. Without this `Name` is used in place of `DisplayName`. Omitting `DisplayName` altogether would avoid translating this string, but we have unit tests that verify all visible properties have a `DisplayName` value.

**Pull request merged**